content/6_Advanced/MinCostFlow.mdx

```diff
@@ -211,19 +211,103 @@ and
 
 ### Assignment Problem
 
-<IncompleteSection>
+The assignment problem is best understood as a MCMF problem. In our formulation,
+we will assign $W$ workers to $J$ jobs, $J \ge W$, and the cost of assigning the
+$i$th job to the $j$th worker is $A_{i,j}$.
 
-Better resources? I honestly just found some that came up in a google search.
+We will create a flow network with a source node $S$, job nodes $J_i$, worker
+nodes $W_j$, and a sink node $E$.
 
-</IncompleteSection>
+We will have the following edges:
+
+- $(S, J_i)$ with $(capacity, cost) = (1, 0)$
+- $(J_i, W_j)$ with $(1, A_{i,j})$
+- $(W_j, E)$ with $(1, 0)$
+
+The answer will be the minimum cost of the max flow of the graph.
+
+To solve this, we will use the MCMF algorithm detailed above. In every iteration
+of the algorithm, we will assign a job to a worker. In each iteration, we will
+first assign the job to an auxilary worker, then we will try to find an
+augmenting path with minimum cost. To optimize this, we will only run Dijkstra
+on the worker nodes. For details, refer to the code.
+
+Since there are $J$ iterations (for each job) and each iteration takes $O(W^2)$
+time, the overall time complexity is $O(J W^2)$.
+
+### Implementation
+
+From Wikipedia (which is then copied from e-maxx):
+
+```cpp
+/**
+ * @return the jobs of each worker in the optimal assignment,
+ * or -1 if the worker is not assigned
+ */
+template <class T> vector<T> hungarian(const vector<vector<T>> &C) {
+	int J = C.size();
+	int W = C[0].size();
```

Comment:
can you rename `C`, `J`, and `W` to something more informative perhaps

Reply:
I think it's pretty informative because it's what I use in the explanation (except for A/C, I'll change that when i get to my laptop)

Reply:
wdym except for ac?

Reply:
I used A in the explanation and C in the code, this will be fixed in the next commit.